feat: HTML/CSS prompt reimplemented (PR #1486)

Jhonattan2121 opened the pull request:
changes :
BackgroundGenerator file re-implemented in ThemeSettingsEditor.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Multi-model background generation with per-model fallback and clearer model-reported responses (returned model included).

* **Refactor**
  * Background generator UI adds a "Checking tokens..." waiting state, updated button/spinner behavior, and is integrated into theme settings.
  * Client-side gating now checks live token balance and NFT holdings before enabling actions.

* **Bug Fixes**
  * Improved handling and reporting of partial/failed generation attempts to avoid hangs.

* **Chores**
  * On-chain contract addresses updated for SPACE and NFT checks.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Jhonattan2121 committed:
feat: HTML/CSS prompt reimplemented (PR #1486)
- Updates background API for multiple models
- Refactors BackgroundGenerator and integrations it into theme editors
- Adjusts CodeTabContent and FontsTabContent for the new flow

willyogo commented:
@Jhonattan2121 can you figure out why the token gate for the HTML/CSS background generation isn't successfully detecting nOGs or $SPACE holdings?

It should trigger the check when I try to generate. I have both $SPACE and nOGs in my wallet, but the app thinks I don't.

It's likely related to changes made in this PR that this PR reverts: https://github.com/Nounspace/nounspace.ts/pull/1509 (this PR removed the token gate check from the login flow and deferred it to when a user tries to use a premium feature like background generation)

<img width="257" height="256" alt="image" src="https://github.com/user-attachments/assets/f77c45ce-7f9c-4973-b21e-fda04d1dd7c7" />

Jhonattan2121 committed:
fix: improve token gate UX for background generation

- Button now only shows Checking tokens... after user clicks to generate, not on initial load
- Gate state is fully reactive to token and nOGs changes
- Prevents false blocking and ensures correct feedback for users with required tokens

Jhonattan2121 committed:
fix: button gating considers  or nOGs NFT

Jhonattan2121 committed:
Merge branch 'canary' into HTML/CSS-prompt-re-implemented

Jhonattan2121 committed:
fix: ensures that the useBalance hook only runs on the client-side, preventing SSR from breaking the SPACE balance in production

Jhonattan2121 committed:
fix: correction of space and nogs contract

Jhonattan2121 committed:
fix: adjustments to automatic nOG checking and debug improvements

Jhonattan2121 committed:
Merge branch 'canary' into HTML/CSS-prompt-re-implemented

Jhonattan2121 committed:
Merge branch 'canary' into HTML/CSS-prompt-re-implemented

Jhonattan2121 committed:
fix: correction address

sktbrd committed:
Merge branch 'canary' into HTML/CSS-prompt-re-implemented
